extensions/amp-brightcove/0.1/amp-brightcove.js

```diff
@@ -315,63 +315,44 @@ class AmpBrightcove extends AMP.BaseElement {
    */
   getIframeSrc_() {
     const {element: el} = this;
-
     const account = userAssert(
       el.getAttribute('data-account'),
       'The data-account attribute is required for <amp-brightcove> %s',
       el
     );
-
-    const embed = el.getAttribute('data-embed') || 'default';
-
     this.playerId_ =
       el.getAttribute('data-player') ||
       el.getAttribute('data-player-id') ||
       'default';
 
-    let src =
-      `https://players.brightcove.net/${encodeURIComponent(account)}` +
-      `/${encodeURIComponent(this.playerId_)}` +
-      `_${encodeURIComponent(embed)}/index.html` +
-      '?amp=1' +
-      // These are encodeURIComponent'd in encodeId_().
-      (el.getAttribute('data-playlist-id')
-        ? '&playlistId=' + this.encodeId_(el.getAttribute('data-playlist-id'))
-        : el.getAttribute('data-video-id')
-        ? '&videoId=' + this.encodeId_(el.getAttribute('data-video-id'))
-        : '');
-
-    const customReferrer = el.getAttribute('data-referrer');
-
-    if (customReferrer) {
-      src = addParamToUrl(
-        src,
-        'referrer',
-        this.urlReplacements_.expandUrlSync(customReferrer)
-      );
-    }
-
+    const urlParams = {};
     if (this.consentState_) {
-      src = addParamToUrl(src, 'ampInitialConsentState', this.consentState_);
+      urlParams['ampInitialConsentState'] = this.consentState_;
     }
     if (this.consentSharedData_) {
-      src = addParamToUrl(
-        src,
-        'ampConsentSharedData',
-        JSON.stringify(this.consentSharedData_)
+      urlParams['ampConsentSharedData'] = JSON.stringify(
+        this.consentSharedData_
       );
     }
     if (this.consentString_) {
-      src = addParamToUrl(src, 'ampInitialConsentValue', this.consentString_);
+      urlParams['ampInitialConsentValue'] = this.consentString_;
     }
 
     el.setAttribute('data-param-playsinline', 'true');
+    el.removeAttribute('data-param-autoplay');
 
-    if (el.hasAttribute('data-param-autoplay')) {
-      el.removeAttribute('data-param-autoplay');
-    }
     // Pass through data-param-* attributes as params for plugin use
-    return addParamsToUrl(src, getDataParamsFromAttributes(el));
+    return getBrightcoveIframeSrc(
+      account,
+      this.playerId_,
+      el.getAttribute('data-embed') || 'default',
+      el.getAttribute('data-playlist-id'),
+      el.getAttribute('data-video-id'),
+      el.hasAttribute('data-referrer')
+        ? this.urlReplacements_.expandUrlSync(el.getAttribute('data-referrer'))
+        : undefined,
+      {...urlParams, ...getDataParamsFromAttributes(el)}
+    );
```

Comment:
nit: this might read more nicely and minify better using `Element.dataset`. Note that `dash-case` becomes `camelCase`.

```suggestion
  const {
    'embed': embed = "default",
    'playlistId': playlistId,
    'videoId': videoId,
    'referrer': referrer,
  } = el.dataset;
  return getBrightcoveIframeSrc(
    account,
    this.playerId_,
    embed,
    playlistId,
    videoId,
    referrer != null
      ? this.urlReplacements_.expandUrlSync(referrer)
      : referrer,
    { ...urlParams, ...getDataParamsFromAttributes(el) }
  );
```